### PR TITLE
Deprecate collection

### PR DESCRIPTION
**Pull request by 0xFelix**
##### SUMMARY

Deprecate this collection and recommend the use of its successor kubevirt.core.

##### ISSUE TYPE

- Docs Pull Request

### CODE REVIEW

**Comment by 0xFelix** [comment]
Ping @Andersson007

File: README.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # KubeVirt Collection for Ansible
+
+THIS COLLECTION IS DEPRECATED, USE [kubevirt.core](https://github.com/kubevirt/kubevirt.core) INSTEAD!
```
Comment (Andersson007) [inline README.md]:
```suggestion
THIS COLLECTION IS **UNMAINTAINED** AND DOES NOT WORK CORRECTLY WITH THE LATEST VERSIONS OF ANSIBLE!. Take a look if a replacing collection [kubevirt.core](https://github.com/kubevirt/kubevirt.core) provides functionality you need. If you're interested in fixing and maintaining `community.kubevirt`, please respond in the [issue](https://github.com/ansible-collections/community.kubevirt/issues/27) but we highly recommend you to consider contributing to [kubevirt.core](https://github.com/kubevirt/kubevirt.core) instead.
```

**Review by Andersson007 — COMMENTED** [review]
@0xFelix hi, thanks for the PR

I discussed the case with the community.
As this collection contains 6 modules and kubevirt.core contains only one, there are no replacements for each module in the new one, so community.kubevirt shouldn't be deprecated now.
When kubevirt.core provides all features that community.kubevirt provides, we can deprecate the latter.

**Comment by 0xFelix** [comment]
Hi @Andersson007,

thanks for the reply.

`kubevirt.core` is going with a different approach than `community.kubevirt` and tries to be intentionally very lean. It does not want to put a lot abstractions on top or provide modules for features no one is going to use (anymore). At least `kubevirt_preset` and `kubevirt_template` are modules for deprecated features, so we don't see a point in providing these modules anymore. Also `kubevirt_pvc` looks like a very outdated module for a feature called DataVolumes right now. I'm not even sure it still works with a recent KubeVirt deployment.

Essentially we are left with `kubevirt_cdi_upload`, which can easily be replaced with `virtctl` and with `kubevirt_rs`, for which we don't have a replacement right now.

Since the collection is unmaintained and mostly broken I'd still vote to deprecate it, so new users looking for an Ansible collection are not unnecessarily confused.

What do you think?

**Comment by Andersson007** [comment]
@0xFelix thanks for explaining!
Not being a specialist in the area, i can't have a strong opinion if the features provided by this collection are not relevant for everyone anymore.
I think having the modified version of the sentence in README (which will appear on Galaxy collection's page after a dummy release), would be a good compromise.

Moreover, you can submit kubevirt.core for inclusion, see https://github.com/ansible-collections/ansible-inclusion#submission-process. It's fine even if it has only one module.
When it gets included, users will get it with `pip install ansible` and its docs will appear on docs.ansible.com. As there are not docs for community.kubevirt anymore and its not present in the package, i don't expect many users will bump into it.

**Comment by 0xFelix** [comment]
@Andersson007 

I'm concerned that users will be confused by picking the wrong collection when looking at the Ansible Galaxy search:

https://galaxy.ansible.com/search?deprecated=false&keywords=kubevirt&order_by=-relevance&page=1

Marking it as deprecated on Galaxy would surely help? 

Also, I'm conflicted about marking the collection only as unmaintained, because I do not want to encourage someone picking up these old and broken modules again. Ideally if people miss a feature they should start contributing on `kubevirt.core` instead. IMO deprecating something does not mean there has to be a 1:1 replacement. `kubevirt.core` is a continuiation of the essentials once provided by `community.kubevirt`, while dropping less useful or even deprecated features.

Inclusion into ansible is something we have to look at. Thanks.

**Comment by Andersson007** [comment]
@0xFelix SGTM though i believe we should consult with the community, I've just created a [topic](https://github.com/ansible-community/community-topics/issues/270). Please join the discussion. Thanks

File: README.md (hunk 1 above)
Comment (Andersson007) [reply README.md]: @0xFelix i've just corrected the wording. Could you please take a look and merge if you agree? Then we can proceed with releasing and galaxy deprecated tag.

**Comment by 0xFelix** [comment]
@Andersson007 I updated the PR, thank you.

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@0xFelix thanks!

**Pull request merged**